### PR TITLE
tests(cipher/rsa): add fuzz test to rsa cipher

### PR DESCRIPTION
**Pull request by mcaci**
Hi, in the context of #480 I'm adding the fuzz tests for the rsa algorithm. I tested it with go test ./cipher/rsa/ to test the corpus and go test -fuzz=FuzzRsa ./cipher/rsa/ to run the fuzzer itself.
If that Is that ok for you to add it let me know. Also if you have any feedbacks about my change let me know so that I can adapt it.
Thanks!

### CODE REVIEW

File: cipher/rsa/rsa_test.go [hunk 1]
```diff
@@ -35,30 +35,27 @@ var rsaTestData = []struct {
 	},
 }
 
-func TestEncryptDecrypt(t *testing.T) {
+func testPrecondition(t *testing.T) (int64, int64, int64) {
 	// Both prime numbers
 	p := int64(61)
 	q := int64(53)
```
Comment (tjgurwara99) [inline cipher/rsa/rsa_test.go:41]:
```suggestion
func testPrecondition(t *testing.T) (int64, int64, int64) {
	t.Helper()
	// Both prime numbers
	p := int64(61)
	q := int64(53)
```
mark this as a helper function. That way you won't have to add "Algorithm failed in preamble stage" everytime an error happens.

File: cipher/rsa/rsa_test.go [hunk 2]
```diff
@@ -35,30 +35,27 @@ var rsaTestData = []struct {
...
-
 	n := p * q
-
 	delta := lcm.Lcm(p-1, q-1)
-
 	e := int64(17) // Coprime with delta
-
 	if gcd.Recursive(e, delta) != 1 {
 		t.Fatal("Algorithm failed in preamble stage:\n\tPrime numbers are chosen statically and it shouldn't fail at this stage")
```
Comment (tjgurwara99) [inline cipher/rsa/rsa_test.go:46]:
```suggestion
		t.Fatal("something went wrong: prime numbers are chosen statically and it shouldn't fail at this stage")
```

Comment (tjgurwara99) [reply cipher/rsa/rsa_test.go:46]: Not that big a difference but a meaningful one - the Go test output shouldn't have newlines and/or tabs (not something you did but if you can change it 🙏🏼)

File: cipher/rsa/rsa_test.go [hunk 3]
```diff
@@ -35,30 +35,27 @@ var rsaTestData = []struct {
...
 	}
-
 	d, err := modular.Inverse(e, delta)
-
 	if err != nil {
 		t.Fatalf("Algorithm failed in preamble stage:\n\tProblem with a modular directory dependency: %v", err)
```
Comment (tjgurwara99) [inline cipher/rsa/rsa_test.go:50]:
```suggestion
		t.Fatalf("something went wrong: problem with %q: %v", "modular.Inverse", err)
```

File: cipher/rsa/rsa_test.go [hunk 4]
```diff
@@ -77,3 +74,24 @@ func TestEncryptDecrypt(t *testing.T) {
...
+
+func FuzzRsa(f *testing.F) {
+	for _, rsaTestInput := range rsaTestData {
+		f.Add(rsaTestInput.input)
+	}
+	f.Fuzz(func(t *testing.T, input string) {
+		e, d, n := testPrecondition(t)
+		encrypted, err := Encrypt([]rune(input), e, n)
+		if err != nil {
+			t.Fatalf("Failed to Encrypt test string:\n\tErrMessage: %v", err)
```
Comment (tjgurwara99) [inline cipher/rsa/rsa_test.go:86]:
```suggestion
			t.Fatalf("failed to encrypt string: %v", err)
```

File: cipher/rsa/rsa_test.go [hunk 5]
```diff
@@ -77,3 +74,24 @@ func TestEncryptDecrypt(t *testing.T) {
...
+		}
+		decrypted, err := Decrypt(encrypted, d, n)
+		if err != nil {
+			t.Fatalf("Failed to Decrypt test string:\n\tErrMessage: %v", err)
```
Comment (tjgurwara99) [inline cipher/rsa/rsa_test.go:90]:
```suggestion
			t.Fatalf("failed to decrypt string: %v", err)
```

File: cipher/rsa/rsa_test.go [hunk 6]
```diff
@@ -77,3 +74,24 @@ func TestEncryptDecrypt(t *testing.T) {
...
+		}
+		if decrypted != input {
+			t.Fatalf("With input string %q was expecting %q but actual was %q",
+				input, input, decrypted)
```
Comment (tjgurwara99) [inline cipher/rsa/rsa_test.go:94]:
```suggestion
			t.Fatalf("expected: %q, got: %q", input, decrypted)
```

**Review by tjgurwara99 — CHANGES_REQUESTED** [review]

**Comment by mcaci** [comment]
Hi @tjgurwara99, I pushed all the changes you suggested me, thanks for all your feedbacks, especially the `t.Helper()` I have just learned about it. Let me know if you have other feedbacks and if I properly addressed the comments from the conversations you shared with me. Thanks again!

**Review by tjgurwara99 — APPROVED** [review]

**Review by raklaptudirm — APPROVED** [review]

**Pull request merged**